### PR TITLE
Fix: Ldap users session expires immediately.

### PR DESCRIPTION
**Pull request by stewartm83**
Please review.

### CODE REVIEW

File: auth.go [hunk 1]
```diff
@@ -472,14 +472,11 @@ func (x *Central) Login(username, password string) (sessionkey string, token *To
...
+
 	sessionExpiry := time.Now().Add(x.NewSessionExpiresAfter)
-	if x.PasswordExpiresAfter != 0 {
-		userPasswordExpiry := user.PasswordModifiedDate.Add(x.PasswordExpiresAfter)
-		if userPasswordExpiry.Before(sessionExpiry) {
-			token.Expires = userPasswordExpiry
-		} else {
-			token.Expires = sessionExpiry
-		}
+	userPasswordExpiry := user.PasswordModifiedDate.Add(x.PasswordExpiresAfter)
```
Comment (bmharper) [inline auth.go]: Why did you remove the check for `x.PasswordExpiresAfter != 0`. That seems like a valid thing to keep.

File: auth.go [hunk 2]
```diff
@@ -472,10 +472,11 @@ func (x *Central) Login(username, password string) (sessionkey string, token *To
 		Identity: user.getIdentity(),
 		UserId:   user.UserId,
 	}
+
 	sessionExpiry := time.Now().Add(x.NewSessionExpiresAfter)
-	if x.PasswordExpiresAfter != 0 {
+	if x.PasswordExpiresAfter != 0 && x.ldap == nil {
```
Comment (stewartm83) [inline auth.go]: @bmharper can we, for the moment, just check whether it is an ldap call and skip password expiry check?

Comment (bmharper) [reply auth.go]: I definitely wouldn't say "for the moment". Any code that gets committed is likely to remain there forever. But will this approach work for a hybrid of ldap and IMQS users in the system? This is actually quite a common scenario.
